packages/sycamore-macro/src/inline_props.rs

```diff
@@ -0,0 +1,108 @@
+use proc_macro2::Span;
+use quote::format_ident;
+use syn::punctuated::Punctuated;
+use syn::{
+    Field, GenericParam, Generics, Ident, Path, PathArguments, PathSegment, Token, Type,
+    TypeImplTrait, TypeParam, TypePath, Visibility,
+};
+
+pub fn create_generic_ident(generics: &Generics) -> Ident {
+    format_ident!("__T{}", generics.params.len())
+}
+
+pub fn resolve_type(generics: &mut Generics, ty: Type) -> Type {
+    match ty {
+        Type::ImplTrait(inner) => add_generic(generics, inner),
+        Type::Array(inner) => {
+            let elem = resolve_type(generics, *inner.elem);
+            Type::Array(syn::TypeArray {
+                bracket_token: inner.bracket_token,
+                elem: Box::new(elem),
+                semi_token: inner.semi_token,
+                len: inner.len,
```

Comment:
I think you could probably use the spread syntax to spread `inner` instead of duplicating all the fields. That would probably be a bit cleaner than this.

Same comment goes for other match branches